File: app/src/main/java/io/caster/zhuinden/realmexamples/Cat.java
package io.caster.zhuinden.realmexamples;

import io.realm.RealmObject;
import io.realm.RealmResults;
import io.realm.annotations.Index;
import io.realm.annotations.LinkingObjects;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.Required;

public class Cat
        extends RealmObject {
    @PrimaryKey
    private String guid;

    @Index
    @Required
    private String name;

    @LinkingObjects("cats")
    private final RealmResults<Person> catOf = null;

    public String getGuid() {
        return guid;
    }

    public void setGuid(String guid) {
        this.guid = guid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public RealmResults<Person> getCatOf() {
        return catOf;
    }
}

File: app/src/main/java/io/caster/zhuinden/realmexamples/models/Cat.kt
package io.caster.zhuinden.realmexamples.models

import io.realm.RealmModel
import io.realm.RealmResults
import io.realm.annotations.LinkingObjects
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass

/**
 * Created by Zhuinden on 2018.04.05..
 */
@RealmClass
open class Cat : RealmModel {
    @PrimaryKey
    var name: String? = null

    @LinkingObjects("cats")
    val owners: RealmResults<Person>? = null
}